src/registry/routes/helpers/apply-default-values.ts

```diff
@@ -0,0 +1,21 @@
+import { OcParameter } from '../../../types';
+
+export default function applyDefaultValues(
+  requestParameters: Dictionary<string | number | boolean> = {},
+  expectedParameters: Dictionary<OcParameter> = {}
+): Dictionary<string | number | boolean> {
+  const optionalParametersWithDefaults = Object.entries(
+    expectedParameters
+  ).filter(([, p]) => !p.mandatory && typeof p.default !== 'undefined');
+
+  optionalParametersWithDefaults.forEach(
+    ([expectedParameterName, expectedParameter]) => {
+      const param = requestParameters[expectedParameterName];
+      if (param == null) {
```

Comment:
previous version of code let `undefined` through at the filter stage and for `undefined` paramters it provided defaults. I think this will not provide default is parameter is `undefined`

Reply:
`param == null` checks for both null and undefined
one of the few cases where the double equality operator is useful and predictable

Reply:
oh it's the double equality... I'm wondering if it's just me, but it be easier for me to understand this if we used `===` and had two conditions like before. double equality operator breaks my mind :P it could be a small follow up PR if it's hard for you to change this branch

Reply:
done